elements/timecontrol/src/main.ts

```diff
@@ -161,33 +167,42 @@ export class EOxTimeControl extends LitElement {
       this._newTimeIndex = this.animationValues.length - 1;
     }
 
-    //@ts-ignore
-    this._animationSource.setTileUrlFunction(
+    if (this._originalTileUrlFunction) {
       //@ts-ignore
-      (tileCoord, pixelRatio, projection) => {
-        const src = this._originalTileUrlFunction(
-          tileCoord,
-          pixelRatio,
-          projection
-        );
-        if (this.urlFunction) {
-          return this.urlFunction(src);
+      this._animationSource.setTileUrlFunction(
+        //@ts-ignore
+        (tileCoord, pixelRatio, projection) => {
+          const src = this._originalTileUrlFunction(
+            tileCoord,
+            pixelRatio,
+            projection
+          );
+          if (this.urlFunction) {
+            return this.urlFunction(src);
+          }
+          const searchParams = new URLSearchParams(
+            src.substring(src.indexOf("?"))
+          );
+          searchParams.set(
+            this.animationProperty,
+            this.animationValues[this._newTimeIndex]
+          );
+
+          return (
+            src.substring(0, src.indexOf("?") + 1) + searchParams.toString()
+          );
```

Comment:
construct a [`URL`](https://developer.mozilla.org/en-US/docs/Web/API/URL) instead of manually parsing `src`. It allows to manipulate its `searchParams` directly.